src/modules/esl-utils/dom/visible.ts

```diff
@@ -0,0 +1,48 @@
+import {getNodeName, getParentNode} from './api';
+
+export interface VisibilityOptions {
+  /** Element will be considered invisible if opacity set to '0' */
+  opacity?: boolean;
+  /** Element will be considered invisible if visibility set to 'hidden' */
+  visibility?: boolean;
+  /** Element will be considered invisible if it's not in viewport */
+  viewport?: boolean;
+  /** Element will be considered invisible if it's height or width is '0' */
+  size?: boolean;
+}
+
+/**
+ * Get the parent of the specified element with matching styles
+ * @param el - element for which to get the parent
+ * @param matchStyles - object of styles for comparing
+ */
+export function findParentByStyles(el: HTMLElement, matchStyles: Partial<CSSStyleDeclaration>): HTMLElement | undefined {
```

Comment:
Method placement doesn't look fine
If we make it public, we should consider some other place for it (e.g., traversing)
Also we need to cover such things by tests if possible